packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateForm.tsx

```diff
@@ -28,27 +46,37 @@ export type PaymentUpdateAuthServerAPIs = Pick<
 >;
 
 export type PaymentUpdateFormProps = {
+  plan: Plan | null;
```

Comment:
This is because `planForId` can return null, https://github.com/mozilla/fxa/blob/44bbb161d958f084c1bb39902554f69a2333de90/packages/fxa-payments-server/src/routes/Subscriptions/index.tsx#L331

Reply:
What I mean is that I don't understand the scenario in which `plan` would be `null`. Now that I'm thinking about it more, I suppose it could be possible that someone has multiple subscriptions, and in that case perhaps `plan` would be `null` for this form, which covers the payment method for all subscriptions?

Reply:
hmmmm if I understand this correctly, we want a plan to determine the currency since we need that for PayPal.  But, when the plan is null, we default to USD.  Since we are dealing with a current customer, it's possible that the customer's subscriptions are not in USD.  And as far as I know, having a currency mismatch between the payment provider and the subscriptions is undefined behavior.

I think we shouldn't allow plan to be null here.

Reply:
@biancadanforth Responding to https://github.com/mozilla/fxa/pull/7909#issuecomment-810391433 here because it helps with context.

Yes, the plan wont' be null as far as I can tell.  I mainly worry about us defaulting USD.  Do you think asserting the Plan is not null is enough?

> When would we expect this error modal -- after the customer clicks the "Change" or "Manage" button in the "Billing Information" section, when the Subscription Management page loads,...?

It'd be after the customer click the button.  (But I don't know if the subscription management page would load without the plans.  I have not verify that.)

Reply:
As long as we don't send a request with a (potentially) mismatching currency, I'm open to handling this in any way that others think is more appropriate.

Reply:
> It'd be after the customer click the button. (But I don't know if the subscription management page would load without the plans. I have not verify that.)

Thanks for clarifying. I tried this out, and we get the "Plan not found" modal when loading the Subscription Management page, so this is not a scenario we should get to.

I also found an existing bug where clicking the X button on that modal just reloads the same page and thus same error modal... I'll file it.

Reply:
> Yes, the plan wont' be null as far as I can tell.

I'm actually not going to file a bug for plan being `null`, now that I re-read this thread and think about it some more. It shouldn't be `null`. I'd rather not try and overthink a scenario that may not even be possible to hit in practice.